cardano-cli/src/Cardano/CLI/Json/Friendly.hs

```diff
@@ -234,39 +241,46 @@ friendlyTxBodyImpl
-                                  friendlyLedgerProposals cOnwards $ toList lProposals
-                          )
-                          era
-                       )
-                 , "voters"
-                    .= ( inEonForEra
-                          Null
-                          ( \cOnwards ->
-                              case txVotingProcedures of
-                                Nothing -> Null
-                                Just (Featured _ TxVotingProceduresNone) -> Null
-                                Just (Featured _ (TxVotingProcedures votes _witnesses)) ->
-                                  friendlyVotingProcedures cOnwards votes
-                          )
-                          era
-                       )
-                 , "currentTreasuryValue" .= toJSON (unFeatured <$> txCurrentTreasuryValue)
-                 , "treasuryDonation" .= toJSON (unFeatured <$> txTreasuryDonation)
                  ]
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    ( \cOnwards ->
+                        case txProposalProcedures of
+                          Nothing -> []
+                          Just (Featured _ TxProposalProceduresNone) -> []
+                          Just (Featured _ (TxProposalProcedures lProposals _witnesses)) ->
+                            ["governance actions" .= (friendlyLedgerProposals cOnwards $ toList lProposals)]
+                    )
+                    era
+                 )
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    ( \cOnwards ->
+                        case txVotingProcedures of
+                          Nothing -> []
+                          Just (Featured _ TxVotingProceduresNone) -> []
+                          Just (Featured _ (TxVotingProcedures votes _witnesses)) ->
+                            ["voters" .= friendlyVotingProcedures cOnwards votes]
+                    )
+                    era
+                 )
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    (const ["currentTreasuryValue" .= toJSON (unFeatured <$> txCurrentTreasuryValue)])
+                    era
+                 )
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    (const ["treasuryDonation" .= toJSON (unFeatured <$> txTreasuryDonation)])
+                    era
+                 )
           )
    where
     friendlyLedgerProposals
       :: ConwayEraOnwards era -> [L.ProposalProcedure (ShelleyLedgerEra era)] -> Aeson.Value
     friendlyLedgerProposals cOnwards proposalProcedures =
       Array $ Vector.fromList $ map (friendlyLedgerProposal cOnwards) proposalProcedures
+    caseByronToBabbageOrConwaysEraOnwards :: (ConwayEraOnwards era -> [a]) -> CardanoEra era -> [a]
+    caseByronToBabbageOrConwaysEraOnwards f =
+      caseByronOrShelleyBasedEra
+        []
+        (caseShelleyToBabbageOrConwayEraOnwards (const []) f)
```

Comment:
That's doing the same same as 
```
monoidForEraInEonA
  :: ()
  => Eon eon
  => Applicative f
  => Monoid a
  => CardanoEra era
  -> (eon era -> f a)
  -> f a
```

Reply:
Nope, because `Aeson.Value` is not a monoid :slightly_smiling_face: (besides, I'm not sure the added abstraction would be worth the harder-to-understand code).